doc/rst/technotes/overloadSets.rst

```diff
@@ -0,0 +1,58 @@
+.. _readme-overload-sets:
+
+======================
+Checking Overload Sets
+======================
+
+The Chapel compiler provides an initial implementation of the
+Overload Sets feature. It is discussed in `CHIP 20
+<https://github.com/chapel-lang/chapel/blob/master/doc/rst/developer/chips/20.rst>`_
+as a solution to "hijacking" problems where program behavior can change
+unexpectedly when imported modules change.
+
+An "overload set" is a group of functions with the same name declared
+in the same module. Overload sets checking is performed at each
+function call. It verifies that all candidate functions identified
+while resolving this call are defined in the same module. See the
+"Function Resolution" section of the language spec for
+resolution-related terminology.
+
+Additionally, if the call resolves to "ref-pair" overloads,
+i.e. for ref, const-ref, and/or value return intent,
+these overloads must be defined in the same module.
+
+The current implementation does not report an error in the following cases:
+
+- The best candidate is "more visible" than the other candidates.
+  That is, it shadows the other candidates along all module import paths
+  starting at the call.
+
+- If it is a method call on a class, the best candidate is defined
+  in the same module as the class of the receiver actual argument.
+
+- The call already generates an "unresolved" or "ambiguous" error.
+
+- The candidates defined in internal modules are ignored, as if
+  they are not present among the identified candidates.
+
+In some cases utilizing function definitions from multiple overload sets
+is actually desirable. To support this, we plan to provide a way
+to merge overload sets. Merging is discussed in `GitHub issue 12635
+<https://github.com/chapel-lang/chapel/issues/12635>`_.
+
+The compiler provides the option ``--no-overload-sets-checks``
+to work around the overload set errors when they are undesirable.
+
+We are evaluating this feature and invite user feedback.
+Our open questions include:
+
+- Should an overload set contain only functions that are declared
+  in the same scope, or some other granularity than a module?
```

Comment:
These seem like completely different questions.

Maybe this would be better:

"Should overload sets continue to consider only module scoping, or should they be updated to consider scopes inside functions for example?"